tests/mixins/test_uploads.py

```diff
@@ -47,6 +48,26 @@ def test_upload_song(self, config, yt_auth):
         response = yt_auth.upload_song(get_resource(config["uploads"]["file"]))
         assert response.status_code == 409
 
+    def test_upload_song_and_verify(self, config, yt_auth):
+        """Upload a song and verify it can be retrieved after it finishes processing."""
+        response = yt_auth.upload_song(get_resource(config["uploads"]["file"]))
+        if not isinstance(response, str) and response.status_code == 409:
+            raise AssertionError(f"{config['uploads']['file']!r} was already uploaded. Delete it and re-run.")
```

Comment:
This is not something that is reproducible. This test would always fail on CI (see `test_upload_song` - it asserts precisely that the return code is 409).

We could do something like the playlist end2end test: delete the uploaded song if it exists (which it will because `test_upload_song` runs before), upload, run your test below.

This would be a great addition because right now we don't actually run the deletion test below - for all I know it could have stopped working two years ago

Reply:
Ok, so now the test seems very robust. I can run it repeatedly without waiting in between runs and it passes consistently, even if the other `test_upload_song` was just ran.

Reply:
In general it looks good to me, but it's failing for me:

```python
yt_auth = <ytmusicapi.ytmusic.YTMusic object at 0x000001B473EACC90>

def test_upload_song_and_verify(self, config, yt_auth: YTMusic):
    """Upload a song and verify it can be retrieved after it finishes processing."""
    upload_response = yt_auth.upload_song(get_resource(config["uploads"]["file"]))
    if not isinstance(upload_response, str) and upload_response.status_code == 409:
        # Song is already in uploads. Delete it and re-upload
        songs = yt_auth.get_library_upload_songs()
        delete_response = None
        for song in songs:
            if song.get("title") in config["uploads"]["file"]:
                delete_response = yt_auth.delete_upload_entity(song["entityId"])
>           assert delete_response == "STATUS_SUCCEEDED"
E           AssertionError: assert None == 'STATUS_SUCCEEDED'

tests\mixins\test_uploads.py:62: AssertionError
```

The other upload test passes.

Reply:
I see, it's because I have a lot of uploaded songs and you didn't set `limit=None`.